scripts/cluster_buildings.py

```diff
@@ -2,152 +2,144 @@
+        input_filepath, crs, house_area_limit
+    )
+    # microgrid_buildings = gpd.read_file(input_filepath)
+    centroids_building = [
+        (row.geometry.centroid.x, row.geometry.centroid.y)
+        for row in microgrid_buildings.itertuples()
+    ]
+    microgrid_buildings["centroid_coordinates"] = centroids_building
+
+    centroids_building = np.array(centroids_building)
+    kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(centroids_building)
     centroids = kmeans.cluster_centers_
 
-    # Select most central point in each cluster
     central_points = []
     for i in range(kmeans.n_clusters):
-        cluster_points = points[kmeans.labels_ == i]
+        cluster_points = centroids_building[kmeans.labels_ == i]
         distances = np.linalg.norm(cluster_points - centroids[i], axis=1)
         central_point_idx = np.argmin(distances)
         central_points.append(cluster_points[central_point_idx])
 
-    # Create GeoJSON feature for each central point
-    features = []
+    central_features = []
     for i, central_point in enumerate(central_points):
-        feature = {
-            "type": "Feature",
-            "geometry": {"type": "Point", "coordinates": central_point.tolist()},
-            "properties": {"cluster": i},
-        }
-        features.append(feature)
-
-    # Create GeoJSON object with all central point features
-    geojson = {"type": "FeatureCollection", "features": features}
-
-    # Write central point GeoJSON data to file
-    with open(output_filepath, "w") as f:
-        json.dump(geojson, f)
+        central_features.append(
+            {
+                "geometry": Point(central_point),
+                "cluster": i,
+            }
+        )
+    central_features = gpd.GeoDataFrame(central_features)
+    central_features.to_file(output_filepath)
 
 
 def get_central_points_geojson_with_buildings(
-    input_filepath, output_filepath, n_clusters
+    input_filepath, output_filepath, n_clusters, crs, house_area_limit
 ):
-    # Load GeoJSON data from file
-    with open(input_filepath, "r") as f:
-        data = json.load(f)
-
-    # Extract coordinates of all points
-    points = []
-    building_ids = []
-    for feature in data["features"]:
-        if feature["geometry"]["type"] == "Point":
-            points.append(feature["geometry"]["coordinates"])
-            building_ids.append(feature["properties"]["id"])
-
-    # Convert points to NumPy array for use with scikit-learn
-    points = np.array(points)
-
-    # Perform k-means clustering
-    kmeans = KMeans(n_clusters=n_clusters, random_state=0).fit(points)
-
-    # Compute centroids of each cluster
+    cleaned_buildings = buildings_classification(input_filepath, crs, house_area_limit)
+    # cleaned_buildings = gpd.read_file(input_filepath)
+    centroids_building = [
+        (row.geometry.centroid.x, row.geometry.centroid.y)
+        for row in cleaned_buildings.itertuples()
+    ]
```

Comment:
It seems this is repeated. Could you explain why?
What is the difference between the previous function and this one?
Same comment on  cleaned_buildings.geometry.centroid applies

Reply:
I had kept the previous structure with two separate functions, and in order to split the buildings into the various clusters it was necessary to recalculate the position of the centroids, I was thinking, what if I compact these two functions into one function with the two separate outputs?

Reply:
See comment below for proposed function